backend/laundry/api_wrapper.py

```diff
@@ -49,41 +61,28 @@ def parse_a_hall(hall_link):
     detailed = []
 
     try:
-        page = requests.get(
-            hall_link,
-            timeout=60,
-            headers={"Authorization": "Basic Sure-Nothing-Could-Go-Wrong-With-This-HaHa-Not"},
-        )
-        # page = requests.get(hall_link, timeout=60)
-    except (ConnectTimeout, ReadTimeout):
+        request = requests.get(room_request_link, timeout=60, headers=headers)
+        request.raise_for_status()
+    except HTTPError:
         return {"washers": washers, "dryers": dryers, "details": detailed}
 
-    soup = BeautifulSoup(page.content, "html.parser")
-    soup.prettify()
-
-    rows = soup.find_all("tr")
-    for row in rows:
-        cols = row.find_all("td")
-        if len(cols) > 1:
-            machine_type = cols[1].getText()
-            if machine_type == "Washer":
-                washers = update_machine_object(cols, washers)
-            elif machine_type == "Dryer":
-                dryers = update_machine_object(cols, dryers)
-            if machine_type in ["Washer", "Dryer"]:
-                try:
-                    time = int(cols[3].getText().split(" ")[0])
-                except ValueError:
-                    time = 0
-                detailed.append(
-                    {
-                        "id": int(cols[0].getText().split(" ")[1][1:]),
-                        "type": cols[1].getText().lower(),
-                        "status": cols[2].getText(),
-                        "time_remaining": time,
-                    }
-                )
-
+    request_json = request.json()
+
+    for machine in request_json:
+        if machine["isWasher"]:
+            washers = update_machine_object(machine, washers)
+        elif machine["isDryer"]:
+            dryers = update_machine_object(machine, dryers)
+        if machine["isWasher"] or machine["isDryer"]:
+            # TODO: Unsure if remaining seconds and the new status ids work on frontend
+            detailed.append(
+                {
+                    "id": machine["id"],
+                    "type": "washer" if machine["isWasher"] else "dryer",
+                    "status": machine["currentStatus"]["statusId"],
+                    "time_remaining": machine["currentStatus"]["remainingSeconds"],
+                }
+            )
     return {"washers": washers, "dryers": dryers, "details": detailed}
```

Comment:
We should make this more readable. Compute washers, dryers, and detailed seperately. Not at the same time.